code/modules/xenobio2/mob/xeno procs.dm

```diff
@@ -64,6 +64,10 @@ Procs for targeting
 
 			reagent_response = chemreact[R.id]
 
+			if(!reagent_response)
+				temp_chem_holder.reagents.clear_reagents()
+				return 1
```

Comment:
This will remove all reagents instead of properly disregarding erroneous ones.

An alternate approach is to move all of the if checks under an if(reagent_response) check, to prevent runtiming.